common/lib/dependabot/config/update_config.rb

```diff
@@ -12,13 +12,18 @@ def initialize(ignore_conditions: nil, commit_message_options: nil)
       end
 
       def ignored_versions_for(dep)
+        name_normaliser = Dependency.name_normaliser_for_package_manager(dep.package_manager)
         @ignore_conditions.
-          select { |ic| ic.dependency_name == dep.name }. # FIXME: wildcard support
+          select { |ic| name_match?(name_normaliser.call(ic.dependency_name), name_normaliser.call(dep.name)) }.
```

Comment:
Idea: `name_normaliser.call(dep.name)` will always have the same value, we could save a few nanos by memoizing outside of the loop.

Reply:
Idea: should the `IgnoreCondition` be what decides if the ignore condition matches this dependency?
Something like `IgnoreCondition#applies_to?(dependency)` or `IgnoreCondition#applies_to?(dependency_name)` 

Meh, just brainstorming.

Reply:
Noted bellow, going to merge it into UpdateConfig and then apply to your branch and refactor from there